xxh_x86dispatch.c

```diff
@@ -572,25 +560,40 @@ typedef struct {
     XXH3_dispatchx86_hashLong64_withSeed   hashLong64_seed;
     XXH3_dispatchx86_hashLong64_withSecret hashLong64_secret;
     XXH3_dispatchx86_update                update;
-} dispatchFunctions_s;
+} XXH_dispatchFunctions_s;
 
-static dispatchFunctions_s g_dispatch = { NULL, NULL, NULL, NULL};
+#define XXH_NB_DISPATCHES 4
 
-#define NB_DISPATCHES 4
-static const dispatchFunctions_s k_dispatch[NB_DISPATCHES] = {
-        /* scalar */ { XXHL64_default_scalar, XXHL64_seed_scalar, XXHL64_secret_scalar, XXH3_64bits_update_scalar },
-        /* sse2   */ { XXHL64_default_sse2,   XXHL64_seed_sse2,   XXHL64_secret_sse2,   XXH3_64bits_update_sse2 },
-#ifdef XXH_DISPATCH_AVX2
-        /* avx2   */ { XXHL64_default_avx2,   XXHL64_seed_avx2,   XXHL64_secret_avx2,   XXH3_64bits_update_avx2 },
+/*!
+ * @internal
+ * @brief Table of dispatchers for @ref XXH3_64bits().
+ *
+ * @pre The indices must match @ref XXH_VECTOR_TYPE.
+ */
+static const XXH_dispatchFunctions_s XXH_kDispatch[XXH_NB_DISPATCHES] = {
+#if XXH_DISPATCH_SCALAR
+    /* Scalar */ { XXHL64_default_scalar, XXHL64_seed_scalar, XXHL64_secret_scalar, XXH3_update_scalar },
 #else
-        /* avx2 */ { NULL, NULL, NULL, NULL },
+    /* Scalar */ { NULL, NULL, NULL, NULL },
 #endif
-#ifdef XXH_DISPATCH_AVX512
-        /* avx512 */ { XXHL64_default_avx512, XXHL64_seed_avx512, XXHL64_secret_avx512, XXH3_64bits_update_avx512 }
+    /* SSE2   */ { XXHL64_default_sse2,   XXHL64_seed_sse2,   XXHL64_secret_sse2,   XXH3_update_sse2 },
+#if XXH_DISPATCH_AVX2
+    /* AVX2   */ { XXHL64_default_avx2,   XXHL64_seed_avx2,   XXHL64_secret_avx2,   XXH3_update_avx2 },
 #else
-        /* avx512 */ { NULL, NULL, NULL, NULL }
+    /* AVX2   */ { NULL, NULL, NULL, NULL },
+#endif
+#if XXH_DISPATCH_AVX512
+    /* AVX512 */ { XXHL64_default_avx512, XXHL64_seed_avx512, XXHL64_secret_avx512, XXH3_update_avx512 }
+#else
+    /* AVX512 */ { NULL, NULL, NULL, NULL }
 #endif
 };
+/*!
+ * @internal
+ * @brief The selected dispatch table for @ref XXH3_64bits().
+ */
+static XXH_dispatchFunctions_s XXH_g_dispatch = { NULL, NULL, NULL, NULL };
```

Comment:
minor : more a comment : 
having a `XXH_*` prefix in front of every symbol and name
is important to mitigate namespace pollution,
particularly for source code which will be included into some 3rd-party source code, like a `*.h`.
But here, `xxh_x86dispatch.c` is not designed to be included,
so this additional namespace precaution feels superfluous.

not a big issue either : it's not detrimental.